### ISSUE TITLE
Setting ModifiedProperties does not update individual entity properties

### ISSUE THREAD

**Issue by noufionline** [TrackableEntities/TrackableEntities.Core#18]
I am running your sample and as I press the key to update, I can see that only unit price is updated for the first record. But if you look at the below SQL generated, the whole order details table is updated.

```SET NOCOUNT ON;

--//////////////////////////////////////////////////

DELETE FROM [OrderDetail]
WHERE  [OrderDetailId] = 13161 /* @p0 */;

--//////////////////////////////////////////////////

SELECT @@ROWCOUNT;

--//////////////////////////////////////////////////

UPDATE [OrderDetail]
SET    [Discount] = 0 /* @p1 */,
       [OrderId] = 22079 /* @p2 */,
       [ProductId] = 1 /* @p3 */,
       [Quantity] = 5 /* @p4 */,
       [UnitPrice] = 11 /* @p5 */
WHERE  [OrderDetailId] = 13160 /* @p6 */;

--//////////////////////////////////////////////////

SELECT @@ROWCOUNT;

--//////////////////////////////////////////////////

INSERT INTO [OrderDetail]
            ([Discount],
             [OrderId],
             [ProductId],
             [Quantity],
             [UnitPrice])
VALUES      (0 /* @p7 - [Discount] */,
             22079 /* @p8 - [OrderId] */,
             3 /* @p9 - [ProductId] */,
             15 /* @p10 - [Quantity] */,
             30 /* @p11 - [UnitPrice] */);

--//////////////////////////////////////////////////

SELECT [OrderDetailId]
FROM   [OrderDetail]
WHERE  @@ROWCOUNT = 1
       AND [OrderDetailId] = scope_identity();
```

**Comment by noufionline** [comment]
I also tried this test but the change track is not working for me here as well.

```csharp

    public class ModifiedPropertiesTest
    {
        private readonly NorthwindDbContextFixture _fixture;

        public ModifiedPropertiesTest()
        {
            _fixture = new NorthwindDbContextFixture();
            _fixture.Initialize(false);
            HibernatingRhinos.Profiler.Appender.EntityFramework.EntityFrameworkProfiler.Initialize();
        }


        [Fact(DisplayName = "Only modified properties are updated")]
        public void Only_modified_properties_are_updated()
        {
            //Arrange
            var context = _fixture.GetContext();
            //Act
            var product = context.Products.First();
            product.UnitPrice = 1111;
            product.ModifiedProperties = new List<string> {"UnitPrice"};
            product.TrackingState = TrackingState.Modified;
            //Assert
            context.ApplyChanges(product);
            context.SaveChanges();

        }
    }
```


And here is then result from profiler

```

SET NOCOUNT ON;

--//////////////////////////////////////////////////

UPDATE [Products]
SET    [CategoryId] = 1 /* @p0 */,
       [Discontinued] = 0 /* @p1 */,
       [ProductInfoKey1] = NULL /* @p2 */,
       [ProductInfoKey2] = NULL /* @p3 */,
       [ProductName] = 'Test Product 1' /* @p4 */,
       [PromoId] = NULL /* @p5 */,
       [UnitPrice] = 1111 /* @p6 */
WHERE  [ProductId] = 1 /* @p7 */;

--//////////////////////////////////////////////////

SELECT @@ROWCOUNT;
```

**Comment by tonysneed** [comment]
I'll incorporate the unit test and fix (#19) in the next release.
